File: test-suite/src/test/java/io/micronaut/docs/aws/sdk/v2/S3ClientBuilderListener.java
package io.micronaut.docs.aws.sdk.v2;

import io.micronaut.context.annotation.Requires;
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import jakarta.inject.Singleton;
import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
import software.amazon.awssdk.core.retry.RetryMode;
import software.amazon.awssdk.services.s3.S3ClientBuilder;

@Requires(property = "spec.name", value = "S3ClientBuilderTest")
//tag::listener[]
@Singleton
public class S3ClientBuilderListener implements BeanCreatedEventListener<S3ClientBuilder> {

    @Override
    public S3ClientBuilder onCreated(BeanCreatedEvent<S3ClientBuilder> event) {
        S3ClientBuilder builder = event.getBean();
        builder.overrideConfiguration(ClientOverrideConfiguration.builder().retryPolicy(RetryMode.LEGACY).build());

        return builder;
    }
}
//end::listener[]

File: test-suite-python/src/test/python/micronaut/docs/aws/sdk/v2/S3ClientBuilderListener.py
from jakarta.inject import Singleton
from micronaut.context.annotation import Requires
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener
from software.amazon.awssdk.core.client.config import ClientOverrideConfiguration
from software.amazon.awssdk.core.retry import RetryMode
from software.amazon.awssdk.services.s3 import S3ClientBuilder


@Requires(property="spec.name", value="S3ClientBuilderTest")
# tag::listener[]
@Singleton
class S3ClientBuilderListener(BeanCreatedEventListener[S3ClientBuilder]):

    def onCreated(self, event: BeanCreatedEvent[S3ClientBuilder]) -> S3ClientBuilder:
        builder = event.getBean()
        builder.overrideConfiguration(ClientOverrideConfiguration.builder().retryPolicy(RetryMode.LEGACY).build())

        return builder
# end::listener[]
